src/js/annotations/barplot.js

```diff
@@ -0,0 +1,199 @@
+    var chr, chrAnnots, chrModel, barAnnots, i, annot, px, j, barPx, nextBarPx,
+        barWidth = ideo.config.barWidth;
+
+    for (chr in annots) {
+        chrAnnots = annots[chr].annots;
+        chrModel = chrModels[annots[chr].chr]; // get chr by name
+        barAnnots = bars[chrModel.chrIndex].annots;
+        for (i = 0; i < chrAnnots.length; i++) {
+            annot = chrAnnots[i];
+            px = annot.px - ideo.bump;
+            let var_count = annot["count"];
+            for (j = 0; j < barAnnots.length; j++) {
+                barPx = barAnnots[j].px;
+                nextBarPx = barPx + barWidth;
+                if (j === barAnnots.length - 1) nextBarPx += barWidth;
+                if (px >= barPx && px < nextBarPx) {
+                    bars[chrModel.chrIndex].annots[j].count += var_count;
+                    bars[chrModel.chrIndex].annots[j].annots.push(annot);
+                    break;
+                }
+            }
+        }
+    }
+    return bars;
+}
+
+function setIdeoMaxAnnotsPerBar(bars, isFirstGet, ideo) {
+    var maxAnnotsPerBarAllChrs, i, maxAnnotsPerBar, annots, chr, j, barCount;
+
+    if (isFirstGet || ideo.config.BarplotScaling === 'relative') {
+        maxAnnotsPerBarAllChrs = 0;
+        for (i = 0; i < bars.length; i++) {
+            maxAnnotsPerBar = 0;
+            annots = bars[i].annots;
+            chr = bars[i].chr;
+            for (j = 0; j < annots.length; j++) {
+                barCount = annots[j].count;
+                if (barCount > maxAnnotsPerBar) maxAnnotsPerBar = barCount;
+                if (barCount > maxAnnotsPerBarAllChrs) {
+                    maxAnnotsPerBarAllChrs = barCount;
+                }
+            }
+            ideo.maxAnnotsPerBar[chr] = maxAnnotsPerBar;
+        }
+        ideo.maxAnnotsPerBarAllChrs = maxAnnotsPerBarAllChrs;
+    }
+}
+
+/**
+ * Set each bar's height to be proportional to the height of the bar with the
+ * most annotations
+ */
+function setProportionalBarHeight(bars, ideo) {
+    var i, annots, chr, j, barCount, barCountRatio, height,
+        ideoIsRotated = ideo._layout._isRotated;
+
+    for (i = 0; i < bars.length; i++) {
+        annots = bars[i].annots;
+        chr = bars[i].chr;
+        for (j = 0; j < annots.length; j++) {
+            barCount = annots[j].count;
+            if (ideo.config.BarplotScaling === 'relative') {
+                barCountRatio = barCount / ideo.maxAnnotsPerBar[chr];
+            } else {
+                barCountRatio = barCount / ideo.maxAnnotsPerBarAllChrs;
+            }
+            if (ideoIsRotated === false) {
+                height = barCountRatio * ideo.config.chrMargin;
+            } else {
+                height = barCountRatio * ideo.config.chrHeightOriginal * 3;
+            }
+            if (isNaN(height)) {
+                height = 0;
+            }
+            bars[i].annots[j].height = height;
+        }
+    }
+    return bars;
+}
```

Comment:
This function is identical to the function with the same name in `histogram.js`.  Import that one, don't duplicate it.